internal/utils/utils.go

```diff
@@ -17,20 +17,29 @@ package utils
 import (
 	"context"
 	"fmt"
+	"regexp"
+	"strings"
 	"time"
 
 	"github.com/cilium/cilium-cli/defaults"
 )
 
+func CheckVersion(version string) bool {
+	var check = regexp.MustCompile(`^([v]|0|[1-9][0-9]*\.)?(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(-[a-zA-Z0-9]+)*\.*(0|[1-9][0-9]*)?`).MatchString
```

Comment:
Does this now allow `v.1.0`? I'm not sure about the `[v]|` since the OR operator (`|`) seems like it is selecting between either starting with a v or a 0 or a number series starting with a 1.

Reply:
And for the other question about discouraging `v1.9`, I think that's because the first section here (before the first `?`) may all occur zero or one times due to the `?`. But we probably just want that first section to always occur, so removing the question mark and moving the period literal (`\.`) outside the brackets.